### PR TITLE
Add support for queries in both C# and VbNet

### PR DESCRIPTION
**Pull request by sagi1623**

### CODE REVIEW

**Comment by sagi1623** [comment]
Hello @olafurpg,
I tried to make the indexing work for query syntax. Please review it when you have some time.

File: snapshots/output-net6.0/syntax/Main/QuerySyntax.cs [hunk 1]
```diff
@@ -35,92 +35,187 @@ void Select()
           var x = from a in sourceA select a.Method();
 //            ^ definition local 0
 //              documentation ```cs\n? x\n```
+//                     ^ definition local 1
+//                       documentation ```cs\n? a\n```
```
Comment (olafurpg) [inline snapshots/output-net6.0/syntax/Main/QuerySyntax.cs:39]: Is it expected that the type is `?` in the documentation?

**Review by olafurpg — APPROVED** [review]
This is a great fix! LGTM 👍🏻

File: snapshots/output-net6.0/syntax/Main/QuerySyntax.cs (hunk 1 above)
Comment (sagi1623) [reply snapshots/output-net6.0/syntax/Main/QuerySyntax.cs:39]: I will take a closer look, in some cases roslyn gives not so useful information :)

Comment (sagi1623) [reply snapshots/output-net6.0/syntax/Main/QuerySyntax.cs:39]: I took a look and actually the symbol that `GetDeclaredSymbol` returns does not contain any type information at all. And the question mark comes from the `SymbolDisplayMiscellaneousOptions.IncludeNullableReferenceTypeModifier` option for displaying the symbol.
I think there is not a lot of things we could do at the moment, with the fact that the symbol return by roslyn does not contain any type information.

Comment (olafurpg) [reply snapshots/output-net6.0/syntax/Main/QuerySyntax.cs:39]: Fair enough. Thank you for looking into this and confirming 🙏🏻

**Pull request merged**